src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -2565,7 +2567,7 @@ void AuraEffect::HandleAuraMounted(AuraApplication const* aurApp, uint8 mode, bo
         {
             vehicleId = creatureInfo->VehicleId;
 
-            if (!displayId || vehicleId)
+            if (!displayId)
```

Comment:
Idk why `vehicleId` are needed for get a model. I have remove because he override `displayId `by creature model